src/ccache.cpp

```diff
@@ -3618,7 +3618,7 @@ ccache(Context& ctx, int argc, char* argv[])
 
   initialize(ctx);
 
-  // TODO, now calling init log HERE, not in vlog cc_log_argv
+  // TODO!, now calling init log HERE, not in vlog cc_log_argv
   // i.e. these functions MUST NOT be called until this line runs.
   init_log(ctx.config);
```

Comment:
`init_log()` was called automatically in the first `cc_log()` call. Either init now happens before any log calls or
`Config::debug()` / `Config::log_file()` (these are required in `init_log()` become global variables. The latter would be easier because `warn_log_fail` **also** requires  `Config::log_file()`.

This means Config would parse these variables and then write them into global variables.

Reply:
Not sure if I have the full picture yet, but running `init_log` early instead of at the first `cc_*log*` call makes sense.